### PR TITLE
Restrict ImageReader backed Platform Views to Android 33+

### PR DESCRIPTION
**Pull request by johnmccutchan**
This is necessary to be able to access the fence associated with each frame that is produced. This API is not available below 33.

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java [hunk 1]
```diff
@@ -371,24 +376,36 @@ public void pushImage(Image image) {
...
     public Image acquireLatestImage() {
       Image r;
       synchronized (this) {
         r = this.image;
         this.image = null;
       }
+      if (r != null) {
+        try {
+          SyncFence fence = r.getFence();
+          boolean signaled = fence.awaitForever();
```
Comment (jonahwilliams) [inline shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java:401]: 😢 

Not necessary to do now, but we _might_ be able to get the Vulkan drive to do the waiting via something like https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkExternalFenceHandleTypeFlagBits.html

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java (hunk 1 above)
Comment (johnmccutchan) [reply shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java:401]: I'm not sure it is possible to convert a SyncFence into a file descriptor (although it is possible to serialize into a Parceable and manually extracting the fd -- just unclear on how stable the parceable format is).

Comment (jonahwilliams) [reply shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java:401]: Yeah I'm not sure if that is exactly the right API, just that I thought there was something like that

**Pull request merged**